core/types/transaction.go

```diff
@@ -110,6 +111,7 @@ type TxData interface {
 
 	// Celo specific fields
 	feeCurrency() *common.Address
+	maxFeeInFeeCurrency() *big.Int
```

Comment:
IIRC @piersy suggested that we can avoid extending TxData by checking the tx type and casting to a specific tx type whenever we want to use a tx-type-specific attribute. I think that can work, but for consistency it should either be done for `feeCurrency` too or not at all.
I think going with the current approach is fine for now. I just wanted to bring up the point to see if there are any strong opionions or other thoughs about it.

Reply:
I generally don't like the concept of the TxData interface exposing the union of all struct fields.
If I would contribute to upstream, I would vote to keep those changes as is for the purpose of consistency though. 
When considering minimizing the diff and stronger segregation of Celo specific features for our fork I tend to lean towards the suggestion of @piersy.


So do you suggest implementing this in this PR?

Reply:
No, let's keep it as it is. If we want to change it, we should do so in a separate PR.